src/transformers/trainer.py

```diff
@@ -2553,7 +2552,16 @@ def push_to_hub(self, commit_message: Optional[str] = "add model", **kwargs) ->
         if not self.is_world_process_zero():
             return
 
-        return self.repo.push_to_hub(commit_message=commit_message)
+        git_head_commit_url = self.repo.push_to_hub(commit_message=commit_message)
+        # push separately the model card to be independant from the rest of the model
+        if self.args.should_save:
+            self.create_model_card(model_name=model_name, **kwargs)
+            try:
+                self.repo.push_to_hub(commit_message="update model card README.md")
+            except Exception as exc:
```

Comment:
Not sure whether it's best to catch the error and log it, or just let it be raised naturally. I don't have a strong opinion on this so let's see what others think!

Reply:
Yes me neither IDK, it's an ergonomy issue because at this stage, the training has been pushed, so the question is what should we do with a faulty Readme ?

Reply:
And without forgetting fixing model card metadata generation like with this issue https://github.com/huggingface/transformers/issues/13528

Reply:
@LysandreJik, any thoughts?

Reply:
If this happens mid-training, I would advocate for a very visible warning that the model card is incorrect, rather than erroring out. However, if it's possible to generate the model card before the training/evaluation starts (I understand some values will be invalid such as evaluation results) and identify a potential failure there, then we could error out.

Definitely fine to just log the error though.

PS: Shouldn't the exception caught be a bit more specific than `Exception`?

Reply:
About the warnings on git push. git push warnings are now logged as a logger.warning cf. https://github.com/huggingface/huggingface_hub/pull/326/files
The only possibility here where user could not see the warning is if her/his script using transformers sets the `logLevel` to `error` only. Elsewhere, the user gets the same warning as huggingface-cli users. Am i wrong ?

Reply:
I think logging at the error level is fine. Can you remove the PR from draft and mrge it?

Reply:
Done, please note that I mixed (may be I'm wring) a [commit to update to the docs](https://github.com/huggingface/transformers/pull/13514/commits/5255d1cf99243ca9081c6a50a9b5bcff4d73730d), because [huggingface/model_card](https://github.com/huggingface/model_card/issues/5) is gonna be deprecated soon I think

Reply:
I resolved the conflict by adding **blocking=blocking**. If it's okay for you, please feel free to merge directly